bin/stack_functions

```diff
@@ -103,10 +115,15 @@ function add_faux {
     shift 3
     bin/run_faux $fnum $*
     add_iface $switch $port faux-$fnum
-    while ! docker exec forch-faux-$fnum ip addr show faux-eth0 | fgrep inet; do
+    for i in `seq 1 20`; do
         echo Waiting for IP address...
+        ip_address=$(docker exec forch-faux-$fnum ip addr show faux-eth0 | egrep -o '192\.168\.1\.[0-9]{1,3}' || true)
+        if [[ -n "$ip_address" ]]; then
+            break
+        fi
         sleep 2
     done
+    declare -g "ip_faux_$fnum"="$ip_address"
```

Comment:
ug -- this is a really messy way to do this. Do we know where this came from?  It's messy because it's "hidden" how the value gets set from the perspective of where it's used.

Reply:
So the main reason I added line 126 is that I could not find other ways to return the ip address to the caller of the add_faux function.. maybe I can just directly get the ip address after the function is called.. Will raise another PR.

Reply:
Also, the regex above is pretty ugly -- you're matching on a specific IP prefix which is fragile. I'd recommend something like "inet ([0-9.])/" with sed (as a capture group):

Reply:
`ip addr show wlp0s20f3 | sed -nr 's~inet ([0-9.]+)/.*~\1~p'`